src/cache.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+
+// TODO: move to its own package
+
+import fs from 'fs-extra'
+import moment from 'moment'
+
+type Options = {
+  cacheFn: Function
+}
+
+export default class {
+  static async fetch (cachePath: string, cacheDuration: number, options: Options): Promise<?Array<?string>> {
```

Comment:
why allow nulls here? can't we always return `Promise<Array<string>>`? We can just make it an empty array if needed